### PR TITLE
reduce duplicate assignment logger invocations by instantiating with an LRU cache; use commons v1.7.0

### PR DESCRIPTION
**Pull request by leoromanovsky**
[//]: #  (Link to the issue corresponding to this chunk of work)
Fixes: #__issue__

## Motivation and Context
[//]: #  (Why is this change required? What problem does it solve?)

defaults to using the lru cache to reduce duplicate assignments.

since this is this is a server-side sdk we expect many more keys in this cache from a variety of subjects. to determine a sensible default to decided to arbitrarily limit it to approximately 10 mb of memory - this seems small enough to not bring a lot of attention to itself.

worked with chatgpt to understand the memory usage of `lru-cache`:

Screenshot 2023-10-18 at 11 30 55 PM

Screenshot 2023-10-18 at 11 30 44 PM

## Description
[//]: # (Describe your changes in detail)

## How has this been tested?
[//]: # (Please describe in detail how you tested your changes)

[//]: # (OPTIONAL)
[//]: # (Add one or multiple labels: enhancement, refactoring, bugfix)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,7 +32,7 @@
     "axios": "^0.27.2",
     "http-status-codes": "^2.2.0",
     "lru-cache": "^9.1.1",
-    "@eppo/js-client-sdk-common": "^1.5.2"
+    "@eppo/js-client-sdk-common": "file:../js-client-sdk-common"
```
Comment (leoromanovsky) [inline package.json:35]: waiting to publish a new version; was using this on local development

File: src/index.ts [hunk 2]
```diff
@@ -70,6 +83,11 @@ export async function init(config: IClientConfig): Promise<IEppoClient> {
   );
   clientInstance = new EppoClient(configurationRequestor, poller);
   clientInstance.setLogger(config.assignmentLogger);
+
+  clientInstance.setAssignmentCache(
+    config.assignmentCache ?? new LRUAssignmentCache(ASSIGNMENT_CACHE_DEFAULT_MAX_ENTRIES),
```
Comment (leoromanovsky) [inline src/index.ts:88]: I want people to be able to disable the cache if they want to - looking at this more carefully I believe an incoming value of `undefined` would be ignored, so I do need to re-implement this.

Will also add a test to make sure an `undefined` value for `config.assignmentCache` yields no cache

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -1,6 +1,3 @@
----
-labels: mergeable
----
```
Comment (leoromanovsky) [inline .github/pull_request_template.md:3]: this was just junk

File: .github/workflows/lint-test-sdk.yml [hunk 4]
```diff
@@ -11,7 +11,7 @@ jobs:
       - name: Use Node.js
         uses: actions/setup-node@v1
         with:
-          node-version: '14.x'
+          node-version: '16.x'
```
Comment (leoromanovsky) [inline .github/workflows/lint-test-sdk.yml:14]: node 16 is required by commons

File: package.json [hunk 5]
```diff
@@ -31,8 +31,8 @@
   "dependencies": {
     "axios": "^0.27.2",
     "http-status-codes": "^2.2.0",
-    "lru-cache": "^9.1.1",
-    "@eppo/js-client-sdk-common": "^1.5.2"
+    "lru-cache": "^10.0.1",
```
Comment (leoromanovsky) [inline package.json:34]: might as well update this to match the version in commons

**Review by sameerank — APPROVED** [review]

**Pull request merged**